awesome-doc-code-sections.js

```diff
@@ -709,18 +708,48 @@ class CodeSection extends BasicCodeSection {
 
         console.log(this.parsed_code.ce_options)
 
+        this.#initialize_panels()
         if (this.parsed_code.ce_options.add_in_doc_execution)
-            this.#add_execution_panel()
+            this.fetch_execution()
     }
 
-    #add_execution_panel() {
-
-        let left_panel = this.firstChild
-        if (left_panel === undefined || left_panel.tagName !== 'PRE')
+    #initialize_panels() {
+        this.left_panel = this.firstChild
+        if (this.left_panel === undefined || this.left_panel.tagName !== 'PRE')
             console.error('awesome-doc-code-sections.js:CodeSection::add_execution_panel : ill-formed firstChild')
 
+        // right panel
+        this.right_panel = document.createElement('div')
+        this.appendChild(this.right_panel)
+
+        Misc.apply_css(this.right_panel, {
```

Comment:
Could be a good idea to create a CSS class with those attributs and add directly add this class to your element when you create it.

HTML :

<div class="right-panel"></div> 


CSS:

.right-panel {
            display:    'flex',
            alignItems: 'stretch',
            boxSizing:  'border-box',
            position:   'relative',
            top:        0,
            left:       0,
            width:      '50%',
            margin:     0
}

Reply:
For now, in order to minimalize the boilerplate, I chose to delay (if not avoid) involving a CSS file.
From potential users feedback, it'd be better to keep structurale style inside the JS.
Other style rules are left to the user and considered as customization points.

However, you're right about style decoupling. I'll figure out something to clean this up a bit.